lib/main.dart

```diff
@@ -16,7 +20,18 @@ class MyApp extends StatelessWidget {
   Widget build(BuildContext context) {
     return MaterialApp(
       debugShowCheckedModeBanner: false,
-      home: LoginScreen(),
+      home: StreamBuilder<User>(
+          stream: FirebaseAuth.instance.authStateChanges(),
+          builder: (context, snapshot) {
+            if (snapshot.connectionState == ConnectionState.active) {
+              if (snapshot.hasData) {
+                return MyHomePage();
```

Comment:
In case you are confused, you can directly provide the name here by using
```
if (snapshot.hasData) {
  return MyHomePage(snapshot.data.displayName);
}
```
This way you won't have to make the parameters optional and you'd be able to remove the navigators altogether.